src/server/pkg/storage/fileset/storage.go

```diff
@@ -287,6 +318,92 @@ func (s *Storage) WalkFileSet(ctx context.Context, prefix string, f func(string)
 	})
 }
 
+// SetTTL sets the time-to-live for the prefix. If the prefix contains multiple filesets
+// the ttl will be set on all of them and the soonest expiration time will be returned.
+// if no fileset is found SetTTL returns ErrNoFileSetFound
+func (s *Storage) SetTTL(ctx context.Context, prefix string, ttl time.Duration) (time.Time, error) {
```

Comment:
I think we should be a bit more consistent with the naming conventions for this stuff. So, if we need a function like this (I think we may not if we implement renewal on a prefix), then something like `RenewFileSet` seems more consistent and the last parameter would be `expiresAt`. Also, the scoped function below could be `WithRenewFileSet`.

Reply:
We discussed this offline. `RenewDuring` -> `WithRenewal` to be consistent using `With*` for callback style APIs.  `SetTTL` is more literally what's going on, and could be used to set a TTL on a key without one.  After postgres this will no longer talk to the garbage collector client.